### PR TITLE
Serialize/Deserialize trait implemented in no-std for numerous types

### PR DESCRIPTION
**Pull request by michalkucharczyk**
serde feature was NOT introduced into updated crates.

Required for https://github.com/paritytech/substrate/issues/13334

Companion for: paritytech/substrate#14261

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2923865

File: bridges/primitives/messages/src/lib.rs [hunk 1]
```diff
@@ -39,8 +39,19 @@ pub mod storage_keys;
...
+	Encode,
+	Decode,
+	Clone,
+	Copy,
+	PartialEq,
+	Eq,
+	RuntimeDebug,
+	TypeInfo,
+	MaxEncodedLen,
+	serde::Serialize,
```
Comment (davxy) [inline bridges/primitives/messages/src/lib.rs:52]: Now we need to ser/de this in `no_std`?

Comment (davxy) [reply bridges/primitives/messages/src/lib.rs:52]: (same for the other previously std-gated types )

**Review by davxy — APPROVED** [review]

**Review by melekes — APPROVED** [review]
👍

**Review by serban300 — APPROVED** [review]

**Comment by michalkucharczyk** [comment]
bot fmt

**Pull request merged**